Navigating UI resources (#250)
mautic/developer-documentation-new: docs/design/navigating_resources.rst

Views
=====

Views in Mautic use a consistent naming pattern to help you organize and locate templates efficiently.

- list: View and manage all objects in a table format, such as contacts or campaigns. This page provides an overview and allows you to take actions on multiple items.

> - details: Access detailed information about a single object, such as a contact profile. You see all available data and related actions for the specific item.

Reviewer: ```suggestion
* **details**: access all the information for a single item, such as a Contact profile. This page displays the specific data and actions available for that individual item.
```

        }
    } %}

Changing template structure
"""""""""""""""""""""""""""

Templates power features throughout the platform. When you change a variable or structure in a template, you affect every place that uses it. Consider the impact before making changes.

You have several options:

1. Update all usages to work with the new structure.
2. Create a new template with your changes, and add a deprecation notice to the old template. Use the new template instead of the original.
3. Add support for custom attributes if that satisfies your needs.

.. note::

   Use a platform template instead of custom HTML whenever possible. If a pull request adds a button with raw HTML, reviewers require you to switch to a platform template.

Naming and organization of templates
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When naming templates, it's important to follow the same naming pattern used by the platform. This makes it easier for other developers to understand the purpose of the template and to find it when needed. 

Names must indicate the part of the UI they're responsible for.

Template naming and organization best practices
"""""""""""""""""""""""""""""""""""""""""""""""

> Use consistent and descriptive naming for your templates to make them easy to locate and understand. Organize related partial templates into subfolders to improve clarity, especially when a single view includes multiple partials.

Reviewer: ```suggestion
Use consistent and descriptive names for your templates to make them easy to find and understand. Organize related partial templates into folders to improve clarity, especially when a single view includes multiple partials.
```

Previous template organizations often placed partial templates in the same folder as their parent view, using a leading underscore to identify partials. For example, the ``@MauticPage/Page/`` folder included:

> - list.html.twig

Reviewer: ```suggestion
* ``list.html.twig``
```

- form.html.twig
- details.html.twig
- _list.html.twig

The file ``_list.html.twig`` served as a partial for rendering tables within the list view.

> With the current standard, group all partials for a specific view inside a subfolder named after the parent template. Rename partials to follow a clear pattern, and remove the leading underscore. For instance, in ``@MauticPage/Page/``, organize files as follows:

Reviewer: Follow the current standard by grouping all partials for a specific view inside a folder named after the parent template. Rename partials to follow a clear pattern and remove the leading underscore. For instance, in ``@MauticPage/Page/``, organize files as follows:
